### PR TITLE
Update docs for -J common option

### LINKED ISSUE #2815: Improve the documentation of GMT map projections
GMT map projections are documented in three separate places.

1. The [table of all map projections](https://docs.generic-mapping-tools.org/latest/proj_codes.html) provides a quick view of all GMT map projections. It's pretty useful if users want to use a specific projection but don't remember the projection code. 
2. The [-J description in the gmt manpage]( https://docs.generic-mapping-tools.org/latest/gmt.html#the-common-gmt-options) shows map projection parameters and also explains their meanings.
3. The ["Map Projections" chapter in the cookbook](https://docs.generic-mapping-tools.org/latest/cookbook/map_projections.html): This chapter provides more details of map projections, and also provides maps showing what the projections look like. However, this chapter doesn't show the full syntax of map projections. For example, the **-Jb** section says 
``` 
To define the projection in GMT you need to provide the following information:
- Longitude and latitude of the projection center.
- Two standard parallels.
- Map scale in inch/degree or 1:xxxxx notation (-Jb), or map width (-JB).
```
but it's not clear which parameters are optional.

Usually, I have to look through all the maps in the cookbook, find the one I need, remember its projection code, and then go the gmt manpage or the quick-view table to check which parameters are required.

Here are suggestions to improve the documentation:

1. Decide the style to describe projection syntax (see #2267)
2. In "Map Projection" chapter of the cookbook, add the map projection syntax at the beginning ot each section. The structure of each section thus look like:
    ```
    Albers conic equal-area projection (-Jb -JB)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    The full syntax of the map projection is:

    -Jblon0/lat0/lat1/lat2/scale or -JBlon0/lat0/lat1/lat2/width
    
    - lon0 is ...
    - lat0 is ...

    More details about the projection.
    The projection is developed by xxx, and mostly used for large regions. 
   
    Here is an example.
    
   Example command.

   Example figure. 
    ```
3. The **-J description in the gmt manpage** may be shortened by only including the quick-view table.

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR revises documentation for **-J**:

1. Reformat the [Projections Specifications](https://docs.generic-mapping-tools.org/dev/proj-codes.html) table
a. List the projections in the left most column
b. Shorten the GMT and PROJ columns by moving the parameters to a separate column
c. List both capital and lower-case options for GMT codes
d. Provide links to the cookbook sections rather than the gmt.conf proj table
2. Move the [Projections Specifications](https://docs.generic-mapping-tools.org/dev/proj-codes.html) table to the [cookbook chapter 4](https://docs.generic-mapping-tools.org/dev/cookbook/options.html#option-j) and include at the start of chapter 6
3. Add the projection syntax to the in the cookbook sections in chapters 5 and 6

**PREVIEW LINK**: https://gmt-23mi81zz4-gmt.vercel.app/

This is the general appearance of the proj-codes table is this PR:

image

And this is the formatting for the projections descriptions in the cookbook:

image

Fixes #2815 

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

File: doc/rst/source/cookbook/coordinate-transformations.rst [hunk 1]
```diff
@@ -1,297 +1,330 @@
...
+:math:`(x,y)` domain, and the nature of :math:`f` itself.
 
-Two subsets of linear will be discussed separately; these are a polar
-(cylindrical) projection and a linear projection applied to geographic
-coordinates (with a 360° periodicity in the *x*-coordinate). We
-will show examples of all of these projections using dummy data sets
-created with :doc:`/gmtmath`, a "Reverse
-Polish Notation" (RPN) calculator that operates on or creates table data:
+Two subsets of linear will be discussed separately; these are a polar (cylindrical) projection and a linear projection
+applied to geographic coordinates (with a 360° periodicity in the *x*-coordinate). We will show examples of all of these
```
Comment (meghanrjones) [inline doc/rst/source/cookbook/coordinate-transformations.rst:41]: @PaulWessel, I couldn't find the linear polar (cylindrical) projection discussion. Can you provide some guidance as to where this is in the cookbook?

Comment (PaulWessel) [reply doc/rst/source/cookbook/coordinate-transformations.rst:41]: All the non-geographic stuff is in the Chapter called 5. GMT Coordinate Transformations

**Comment by PaulWessel** [comment]
No, this is all @joa-quim's effort - perhaps he has docs to add.

**Comment by joa-quim** [comment]
No, I don't. I guess that people using EPSGs should already know what it is. We can add a link to
https://spatialreference.org/ref/epsg/ and tell people to search EPSG as well as equivalent proj4 strings there.

**Review by PaulWessel — APPROVED** [review]

**Pull request merged**